test/testdata/test.sh

```diff
@@ -44,4 +44,4 @@ test_project project-v4-multigroup
 test_project project-v4-declarative-v1
 test_project project-v4-config
 test_project project-v4-with-deploy-image
-test_project project-v4-with-metrics
+test_project project-v4-with-grafana
```

Comment:
This one needs to be done with : https://github.com/kubernetes-sigs/kubebuilder/pull/3111

We change the name of the samples.
So, we need to change it also in the tests so we can adequately test the projects in the CI. 
PS. the same is valid for v3 one.